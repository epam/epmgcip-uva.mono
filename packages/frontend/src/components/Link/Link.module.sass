.link
    text-decoration: none
    color: inherit

    &:hover
        cursor: pointer