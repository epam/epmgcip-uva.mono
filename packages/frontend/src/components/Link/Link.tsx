import css from './Link.module.sass'


interface ButtonProps {
  children: string | JSX.Element;
  onClick: () => void;
  className?: string;
}

export const Link = ({ children, onClick, className }: ButtonProps) => {
  const linkClass = className ? `${css.link} ${className}` : css.link;

  return (
    <a className={linkClass} onClick={onClick}>
      {children}
    </a>
  );
};
