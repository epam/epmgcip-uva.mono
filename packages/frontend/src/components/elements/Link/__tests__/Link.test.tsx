import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { Link } from "../Link";

const textMock = 'Test Link';
const childrenMock = <div>{textMock}</div>;
const onClickMock = vi.fn();
const additionalClassMock = 'additionalTestClass';


describe('Testing: Link', () => {
    it('should render', () => {
      const { container } = render(
        <Link
          children={childrenMock}
          onClick={onClickMock}
          className={additionalClassMock}
        />
      );
  
      expect(container).toMatchSnapshot();
    });
  
    it('should call provided function after click', () => {
        render(
            <Link
              children={childrenMock}
              onClick={onClickMock}
              className={additionalClassMock}
            />
          );
  
      const renderedLink = screen.getByText(textMock);
  
      fireEvent.click(renderedLink);
  
      expect(onClickMock).toHaveBeenCalled();
    });
  });