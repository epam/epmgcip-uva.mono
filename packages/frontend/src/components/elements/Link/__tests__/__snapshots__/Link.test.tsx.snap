// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Link > should render 1`] = `
<div>
  <a
    class="_link_43716b additionalTestClass"
  >
    <div>
      Test Link
    </div>
  </a>
</div>
`;
